src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -426,7 +420,9 @@ object Contexts {
   abstract class FreshContext extends Context {
     def setPeriod(period: Period): this.type = { this.period = period; this }
     def setMode(mode: Mode): this.type = { this.mode = mode; this }
+    def setCompilerCallback(callback: CompilerCallback): this.type = { this.compilerCallback = callback; this }
     def setTyperState(typerState: TyperState): this.type = { this.typerState = typerState; this }
+    def setReporter(reporter: Reporter): this.type = setTyperState(typerState.withReporter(reporter))
```

Comment:
That's a useful helper method. I think teher are several occurrences of `setTyperState(ctx.typerState.withReporter(...))` which would profit from that method. Best to do it now.